metashare/consumers.py

```diff
@@ -32,17 +33,46 @@ class PushNotificationConsumer(AsyncJsonWebsocketConsumer):
 
             channel_layer.group_send(group_name, {
                 'type': 'notify',  # This routes it to this handler.
-                'content': json_message,
+                'content': {
+                    'type': str (will map to frontend Redux event),
+                    'payload': {
+                        'originating_user_id': str,
+                        'message': str (error message, optional),
+                    },
+                    'model_name': str,
+                    'id': str,
+                },
             })
         """
         # Take lock out of redis for this message:
         await clear_message_semaphore(self.channel_layer, event)
         if "content" in event:
-            await self.send_json(event["content"])
+            message = await self.hydrate_message(event["content"])
+            await self.send_json(message)
             return
 
+    async def hydrate_message(self, content):
+        content = deepcopy(content)
+        model_name = content.pop("model_name")
+        id = content.pop("id")
+        # We specifically don't want to include the user model, as that
+        # would cause every generic-message to include the user who's
+        # getting the message. It'd just be noise on the wire.
+        if model_name.lower() != "user":
+            try:
+                instance = await self.get_instance(model=model_name, id=id)
+            except ObjectDoesNotExist:
+                pass
+            else:
+                content["payload"]["model"] = await database_sync_to_async(
+                    instance.get_serialized_representation
+                )(self.scope["user"])
```

Comment:
@wlonk Just to be sure... this will _override_ any existing message-extras set as the `model` property, correct?

Reply:
Correct.